cmd/executor/cmd/root.go

```diff
@@ -77,6 +79,11 @@ var RootCmd = &cobra.Command{
 		if err := executor.DoPush(image, opts); err != nil {
 			exit(errors.Wrap(err, "error pushing image"))
 		}
+		if cleanup {
```

Comment:
Instead of having this in `root.go`, I think it would be better placed in [this](https://github.com/GoogleContainerTools/kaniko/blob/master/pkg/executor/build.go#L256) if statement in `build.go`

If opts.Cleanup is `true`, then delete the filesystem before returning the image.